docs/api/js/utils.md

```diff
@@ -1,68 +1,785 @@
 
 ```typescript
-export const ETH_ADDRESS = "0x0000000000000000000000000000000000000000";
+export const ETH_ADDRESS = '0x0000000000000000000000000000000000000000';
 ```
 
-## ABI of zkSync smart contract
+#### ETH token alias on ZkSync Era
 
 ```typescript
-export const ZKSYNC_MAIN_ABI = new utils.Interface(require("../abi/IZkSync.json"));
+export const L2_ETH_TOKEN_ADDRESS = '0x000000000000000000000000000000000000800a';
 ```
 
-## IERC20 interface
+#### Bootloader
 
-Convenient when interacting with native tokens on zkSync.
+```ts
+export const BOOTLOADER_FORMAL_ADDRESS = '0x0000000000000000000000000000000000008001';
+```
 
-```typescript
-export const IERC20 = new utils.Interface(require("../abi/IERC20.json"));
+#### Contract deployer
+
+```ts
+export const CONTRACT_DEPLOYER_ADDRESS = '0x0000000000000000000000000000000000008006';
+```
+
+#### L1 messenger
+
+```ts
+export const L1_MESSENGER_ADDRESS = '0x0000000000000000000000000000000000008008';
 ```
 
-## Encoding paymaster params
+### Gas
 
-Utility method that returns the correctly formed `paymasterParams` object for the common [paymaster flows](../../dev/developer-guides/aa.md#built-in-paymaster-flows).
+#### `DEFAULT_GAS_PER_PUBDATA_LIMIT`
+
+- Use a large amount of gas for signing on layer 2.
+- The amount ensures any reasonable limit is accepted. 
+
+:::info
+- The operator is NOT required to use the actual value and can use any value up to that signed by the user.
+:::
 
 ```typescript
-export function getPaymasterParams(paymasterAddress: Address, paymasterInput: PaymasterInput): PaymasterParams
+export const DEFAULT_GAS_PER_PUBDATA_LIMIT = 50000;
 ```
 
-The definition of the `PaymasterInput` can be found [here](./types.md).
+#### `REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT`
 
-## Useful gas constants
-Currently, there is no method to accurately estimate the required `gasPerPubdataLimit`. That's why for now, it is highly recommended to provide the `DEFAULT_GAS_PER_PUBDATA_LIMIT`. Users are not charged more by providing it.
-Later on it will be possible to query the current recommended limit.
+Any amount is possible for L1->L2 transactions because the cost per gas is adjusted.
 
-```typescript
-const GAS_PER_PUBDATA_BYTE = 17;
+```ts
+export const REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT = 800;
+```
 
-// The large L2 gas per pubdata to sign. This gas is enough to ensure that
-// any reasonable limit will be accepted. Note, that the operator is NOT required to
-// use the honest value of gas per pubdata and it can use any value up to the one signed by the user.
-// In the future releases, we will provide a way to estimate the current gasPerPubdata.
-export const DEFAULT_GAS_PER_PUBDATA_LIMIT = 50000;
+## Functions
+
+### `applyL1ToL2Alias`
+
+Converts the address that submitted a transaction to the inbox on L1 to the `msg.sender` viewed on L2.
+
+#### Inputs
+
+- `address`: sender address as string.
+
+#### Outputs
+
+?? not sure how to describe what's coming out here
```

Comment:
"The `msg.sender` of the L1->L2 transaction is the `address` is a contract that initiated it."

As a remark:
During normal transaction if contract A calls contract B, the `msg.sender` of the call will be A.
During L1->L2 communication if an EOA X calls contract B, the `msg.sender` of the call will be X (the same logic as before).
During L1->L2 communication if a *contract* A calls contract B, the `msg.sender` of the call will be `applyL1ToL2Alias(A)` (the same logic as before).

Reply:
added